keycloak/group.go

```diff
@@ -0,0 +1,118 @@
+package keycloak
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Group struct {
+	Id        string   `json:"id,omitempty"`
+	RealmId   string   `json:"-"`
+	ParentId  string   `json:"-"`
+	Name      string   `json:"name"`
+	Path      string   `json:"path,omitempty"`
+	SubGroups []*Group `json:"subGroups,omitempty"`
+}
+
+/*
+ * There is no way to get a subgroup's parent ID using the Keycloak API (that I know of, PRs are welcome)
+ * The best we can do is use the group's path to figure out what its parents' names are and iterate over all subgroups
+ * until we find it.
+ */
+func (keycloakClient *KeycloakClient) groupParentId(group *Group) (string, error) {
+	// Check the path of the group being passed in.
+	// If there is only one group in the path, then this is a top-level group with no parentId
+	parts := strings.Split(strings.TrimPrefix(group.Path, "/"), "/")
+
+	if len(parts) == 1 {
+		return "", nil
+	}
+
+	groups, err := keycloakClient.listGroups(group.RealmId)
```

Comment:
Would it be possible to use the `search` query param with the path name at `len(parts)-2` and then check if `SubGroups` contains  the child id?

https://www.keycloak.org/docs-api/4.2/rest-api/index.html#_getgroups